### PR TITLE
[APM][OTel] Span links from _sources

### PR DESCRIPTION
**Pull request by crespocarlos**
## Summary

image

### CODE REVIEW

File: x-pack/plugins/observability_solution/apm/server/routes/span_links/get_span_links_details.ts [hunk 1]
```diff
@@ -120,14 +103,17 @@ async function fetchSpanLinksDetails({
     },
   });
 
+  const spanLinksDetails = response.hits.hits
+    .map((hit) => spanLinksDetailsMapping(hit.fields))
+    .filter((hits): hits is NonNullable<SpanLinksDetails> => !!hits);
```
Comment (crespocarlos) [inline x-pack/plugins/observability_solution/apm/server/routes/span_links/get_span_links_details.ts:108]: This needs to be in the mappings. There are a few places where ts complains about objects possibly being `undefined`

File: x-pack/plugins/observability_solution/apm/server/utils/es_fields_mappings/trace.ts [hunk 2]
```diff
@@ -24,24 +24,30 @@ import {
...
-  spanLinkFromSource?: SpanLink[]
-): WaterfallTransaction | WaterfallSpan | undefined => {
+export type TraceWaterfallMapping = ReturnType<typeof traceWaterfallMapping>;
+
+export const traceWaterfallMapping = ({
+  fields,
+  _source,
+}: {
+  fields: Fields;
+  _source?: Transaction | Span;
```
Comment (crespocarlos) [inline x-pack/plugins/observability_solution/apm/server/utils/es_fields_mappings/trace.ts:49]: This is probably a DX friendlier function signature

File: x-pack/plugins/observability_solution/apm/server/utils/es_fields_mappings/span_links.ts [hunk 3]
```diff
@@ -59,18 +61,28 @@ export const spanLinksDetailsMapping = (fields: Fields) => {
   };
 };
 
-// todo: pending #192337
-export const linkedParentsOfSpanMapping = (fields: Fields) => {
```
Comment (crespocarlos) [inline x-pack/plugins/observability_solution/apm/server/utils/es_fields_mappings/span_links.ts:63]: Not used

File: x-pack/plugins/observability_solution/apm/server/utils/es_fields_mappings/trace.ts [hunk 4]
```diff
@@ -24,24 +24,30 @@ import {
...
 import { normalizeValue } from './es_fields_mappings_helpers';
 import type { WaterfallTransaction, WaterfallSpan } from '../../../common/waterfall/typings';
 import type { Fields } from './types';
 import { serviceMapping } from './service';
 
-export const traceDocMapping = (
-  fields: Fields,
-  spanLinkFromSource?: SpanLink[]
-): WaterfallTransaction | WaterfallSpan | undefined => {
+export type TraceWaterfallMapping = ReturnType<typeof traceWaterfallMapping>;
```
Comment (miloszmarcinkowski) [inline x-pack/plugins/observability_solution/apm/server/utils/es_fields_mappings/trace.ts:42]: Is `TraceWaterfallMapping` type used anywhere? Sorry I can't find it.

**Review by miloszmarcinkowski — APPROVED** [review]
Good job. I like your approach with span links :)

**Review by MiriamAparicio — APPROVED** [review]
Just code review, haven't tested manually 
LGTM

**Review by rmyz — APPROVED** [review]
LGTM

**Pull request closed without merging**